src/libs/Middleware/Reauthentication.ts

```diff
@@ -1,33 +1,67 @@
+import Onyx from 'react-native-onyx';
+import redirectToSignIn from '@libs/actions/SignInRedirect';
 import * as Authentication from '@libs/Authentication';
 import Log from '@libs/Log';
 import * as MainQueue from '@libs/Network/MainQueue';
 import * as NetworkStore from '@libs/Network/NetworkStore';
+import type {RequestError} from '@libs/Network/SequentialQueue';
 import NetworkConnection from '@libs/NetworkConnection';
 import * as Request from '@libs/Request';
+import RequestThrottle from '@libs/RequestThrottle';
 import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
 import type Middleware from './types';
 
 // We store a reference to the active authentication request so that we are only ever making one request to authenticate at a time.
 let isAuthenticating: Promise<void> | null = null;
 
+const reauthThrottle = new RequestThrottle();
+
 function reauthenticate(commandName?: string): Promise<void> {
     if (isAuthenticating) {
         return isAuthenticating;
     }
 
-    isAuthenticating = Authentication.reauthenticate(commandName)
+    const reauthRequest = {
+        commandName,
+    };
+    // eslint-disable-next-line rulesdir/prefer-actions-set-data
+    Onyx.set(ONYXKEYS.REAUTHENTICATION_REQUEST, reauthRequest);
```

Comment:
NAB: Previously I thought we would need to persist and read the re-auth request from storage so that re-auth retry would work when the app was stopped and re-started. However, it's not needed because the app will send a read request immediately on start up and trigger the re-auth flow.

Since we never read this value from Onyx, we can just store it in memory.

Reply:
@neil-marcellini @tgolen removed